bigquery/google/cloud/bigquery/job.py

```diff
@@ -819,6 +819,35 @@ def to_api_repr(self):
         """
         return copy.deepcopy(self._properties)
 
+    def fill_from_default(self, default_job_config):
+        """Merge this job config with another one.
+        The other takes precedence with conflicting keys.
+        This is a naive one level merge.
+
+        :type default_job_config: google.cloud.bigquery.job._JobConfig
+        :param default_job_config:
+            The default job config that will be used to fill in self.
+
+        :rtype: :class:`google.cloud.bigquery.job._JobConfig`
+        :returns: A new job config.
+        """
+        if self._job_type != default_job_config._job_type:
+            raise TypeError(
+                "attempted to merge two incompatible job types: "
+                + repr(self._job_type) + ', '
+                + repr(default_job_config._job_type))
+
+        new_job_config = self.__class__()
+        new_job_config._properties = copy.deepcopy(self._properties)
+
+        new_job_properties = new_job_config._properties[self._job_type]
```

Comment:
I don't think we need to restrict to `self._job_type`. We miss a handful of properties if we do this (such a labels).